packages/dds/tree/src/test/simple-tree/utils.ts

```diff
@@ -43,3 +44,114 @@ export function pretty(arg: unknown): number | string {
 	}
 	return JSON.stringify(arg);
 }
+
+const schemaFactory = new SchemaFactory("test");
+
+const deepTreeNode = schemaFactory.objectRecursive("deep-tree-node", {
+	node: schemaFactory.optionalRecursive([() => DeepTreeNode]),
+	value: schemaFactory.optional(schemaFactory.number),
+});
+
+export class DeepTreeNode extends deepTreeNode {}
+
+/**
+ * Make a deep tree with a single leaf value
+ */
+function makeDeepTree(depth: number, leafValue: number): DeepTreeNode {
+	return depth === 1
+		? new DeepTreeNode({ node: undefined, value: leafValue })
+		: new DeepTreeNode({ node: makeDeepTree(depth - 1, leafValue), value: undefined });
+}
+
+/**
+ * generate a deep tree with a single leaf value
+ */
+export function generateDeepSimpleTree(depth: number, leafValue: number): DeepTreeNode {
+	if (depth < 1) {
+		throw new Error("Depth must be greater than 0");
+	}
+
+	return hydrate(DeepTreeNode, makeDeepTree(depth, leafValue));
+}
+
+/**
+ * read the deep tree and return the depth and value of the leaf node
+ */
+export function readDeepSimpleTree(tree: DeepTreeNode): {
+	depth: number;
+	value: number;
+} {
+	let currentNode = tree;
+	let depth = 1;
+
+	while (currentNode.node !== undefined) {
+		depth += 1;
+		currentNode = currentNode.node;
+	}
+
+	const value = currentNode.value ?? 0;
+	return { depth, value };
+}
+
+/**
+ * Update the deep tree with a new value
+ */
+export function writeDeepSimpleTree(tree: DeepTreeNode, newValue: number): void {
+	let currentNode = tree;
+
+	while (currentNode.node !== undefined) {
+		currentNode = currentNode.node;
+	}
+
+	currentNode.value = newValue;
+}
+
+const wideTreeNode = schemaFactory.object("wide-tree-node", {
+	values: schemaFactory.array(schemaFactory.number),
+});
+
+export class WideTreeNode extends wideTreeNode {}
+
+/**
+ * Make a wide tree with a single layer
+ */
+function makeWideTree(width: number, leafValue: number): WideTreeNode {
+	const values = [];
+	for (let i = 0; i < width; i++) {
+		values.push(leafValue);
+	}
+	return new WideTreeNode({ values });
+}
+
+/**
+ * generate a wide tree with a single layer
+ */
+export function generateWideSimpleTree(width: number, leafValue: number): WideTreeNode {
+	return hydrate(WideTreeNode, makeWideTree(width, leafValue));
+}
+
+/**
+ * read the wide tree and return the width and the sum of all value in the leaf node
+ */
+export function readWideSimpleTree(tree: WideTreeNode): {
+	nodesCount: number;
+	sum: number;
+} {
+	const nodesCount = tree.values.length;
+	let sum = 0;
+	for (const value of tree.values) {
+		sum += value;
+	}
+	return { nodesCount, sum };
+}
```

Comment:
I guess the concern in my previous comment applies to this one too. This function iterates over the whole tree, which we can reasonably expect to be more expensive as the tree grows. If instead of reading the whole tree sequentially, we did "random access" with tree.values[tree.values.length - 1], we maybe would expect constant-ish time regardless of the tree width. Again @taylorsw04 , thoughts on which case we want to test? Or both, so we can check if the increase in execution time with tree size is linear or not?